[BUG]Training ContrastiveOutput with `bpr-max` is raising an error

### Bug description
Training a contrastive task with the pairwise loss `bpr-max` in graph mode raises the following issue: 

```        if ndims_mask is None:
>         raise ValueError(
              "Number of mask dimensions must be specified, even if some dimensions"
              " are None.  E.g. shape=[None] is ok, but shape=None is not.")
E         ValueError: in user code:
E         
E             File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/keras/engine/training.py", line 1160, in train_function  *
E                 return step_function(self, iterator)
E             File "/home/runner/work/models/models/merlin/models/tf/losses/pairwise.py", line 75, in call  *
E                 (
E             File "/home/runner/work/models/models/merlin/models/tf/losses/pairwise.py", line 152, in _separate_positives_negatives_scores  *
E                 y_pred_valid_rows = tf.boolean_mask(y_pred, valid_rows_with_positive_mask)
E             File "/opt/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/tensorflow/python/util/traceback_utils.py", line 141, in error_handler  **
```

### Steps/Code to reproduce bug
1. This unit test trains a RetrievalModelV2 model with the pairwise loss `bpr-max` and is raising the reported issue :  

```
import merlin.models.tf as mm
from merlin.schema import Tags
@pytest.mark.parametrize("run_eagerly", [False])
def test_contrastive_output_with_pairwise_loss(ecommerce_data: Dataset, run_eagerly):
    model = mm.RetrievalModelV2(
        query=mm.Encoder(ecommerce_data.schema.select_by_tag(Tags.USER), mm.MLPBlock([2])),
        candidate=mm.Encoder(ecommerce_data.schema.select_by_tag(Tags.ITEM), mm.MLPBlock([2])),
        output=mm.ContrastiveOutput(
            ecommerce_data.schema.select_by_tag(Tags.ITEM_ID),
            negative_samplers="in-batch",
            candidate_name="item",
        ),
    )
    model.compile(run_eagerly=run_eagerly, loss="bpr-max")
    _ = model.fit(ecommerce_data, batch_size=50, epochs=1)
```
### Expected behavior
We should be able to train a model with ContrastiveOutput and pairwise loss (such as `bpr-max`) in graph mode. 

### Environment details
- Merlin version: latest main branch
- Platform: Linux-5.13.0-39-generic-x86_64-with-glibc2.10
- Python version:  3.8.13
- PyTorch version (GPU?): - 
- Tensorflow version (GPU?):  2.9.2 (True)

### Additional context

After investigating the error, it seems this [line](https://github.com/NVIDIA-Merlin/models/blob/abab6e1a63bea64031e6f6a232a9112d6b14dcd7/merlin/models/tf/outputs/contrastive.py#L229) is converting the predictions tensor from
 `TensorShape([None, None])` to `TensorShape([None])`

--- a/merlin/models/tf/losses/pairwise.py
+++ b/merlin/models/tf/losses/pairwise.py
@@ -72,6 +72,10 @@ class PairwiseLoss(Loss, LossRegistryMixin):
         tf.Tensor
             Loss per example
         """
+        tf.assert_equal(tf.rank(y_true), 2, f"Targets must be 2-D tensor (got {y_true.shape})")
+
+        tf.assert_equal(tf.rank(y_pred), 2, f"Predictions must be 2-D tensor (got {y_pred.shape})")
+
         (
             positives_scores,
             negatives_scores,

--- a/merlin/models/tf/outputs/contrastive.py
+++ b/merlin/models/tf/outputs/contrastive.py
@@ -226,7 +226,6 @@ class ContrastiveOutput(ModelOutput):
         # To ensure that the output is always fp32, avoiding numerical
         # instabilities with mixed_float16 policy
         outputs = tf.cast(outputs, tf.float32)
-        outputs = tf.squeeze(outputs)
 
         targets = tf.concat(
             [

--- a/tests/unit/tf/outputs/test_contrastive.py
+++ b/tests/unit/tf/outputs/test_contrastive.py
@@ -190,6 +190,21 @@ def test_contrastive_only_positive_when_not_training(ecommerce_data: Dataset):
     )
 
 
+@pytest.mark.parametrize("run_eagerly", [True, False])
+def test_contrastive_output_with_pairwise_loss(ecommerce_data: Dataset, run_eagerly):
+    model = mm.RetrievalModelV2(
+        query=mm.Encoder(ecommerce_data.schema.select_by_tag(Tags.USER), mm.MLPBlock([2])),
+        candidate=mm.Encoder(ecommerce_data.schema.select_by_tag(Tags.ITEM), mm.MLPBlock([2])),
+        output=mm.ContrastiveOutput(
+            ecommerce_data.schema.select_by_tag(Tags.ITEM_ID),
+            negative_samplers="in-batch",
+            candidate_name="item",
+        ),
+    )
+    model.compile(run_eagerly=run_eagerly, loss="bpr-max")
+    _ = model.fit(ecommerce_data, batch_size=50, epochs=1)
+
+
 def _retrieval_inputs_(batch_size):
     users_embeddings = tf.random.uniform(shape=(batch_size, 5), dtype=tf.float32)
     items_embeddings = tf.random.uniform(shape=(batch_size, 5), dtype=tf.float32)

